### DISCUSSION TITLE
Horizontal scrolling for x-axis of type timestamp

### DISCUSSION THREAD

**Discussion by enjineerMan** [chartjs/chartjs-plugin-zoom#644] [Q&A]
In the documentation, the parameters for scale limits are min, max, and minRange (max-min). In my project, data is added to the database in regular intervals, and I want the graph to start scrolling horizontally after a certain number of points. How can I accomplish this? min and max won't work because the data is constantly moving forward in time. Even a parameter like 'maxRange' wouldn't work because data from two days may appear at the same time, so the most recent timestamp - the oldest timestamp would yield a negative value.

**Comment by kurkle** [comment]
You'll need to update the options when you update the data. Eg:

chart.data.datasets[0].data.push(Math.random());
chart.options.pluigins.zoom.limits.x.max = chart.data.datasets[0].data.length;
chart.update();

> Even a parameter like 'maxRange' wouldn't work because data from two days may appear at the same time, so the most recent timestamp - the oldest timestamp would yield a negative value.

`timestamp` not meaning the number of seconds/milliseconds elapsed since January 1, 1970?

**Comment by enjineerMan** [comment]
Sorry I could've clarfied, the values on my x-axis are in HH:mm:ss

**Comment by kurkle** [comment]
So the x-axis type is `category` and you have labels array created from timestamps and formatted to HH:mm:ss?

**Comment by enjineerMan** [comment]
Sorry I'm not familiar with the category type, and yes
